From 65d6793e240643116037b4bb758e40b7db5cc7ee Mon Sep 17 00:00:00 2001
From: Bernd Edlinger <bernd.edlinger@hotmail.de>
Date: Fri, 17 Jun 2022 10:25:24 +0200
Subject: [PATCH] Fix compile issues in test/v3ext.c with no-rfc3779

There are no ASIdentifiers if OPENSSL_NO_RFC3779 is defined,
therefore the test cannot be compiled.
---
 test/v3ext.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/test/v3ext.c b/test/v3ext.c
index 0d371ec280f1..29e33d16a5fa 100644
--- a/test/v3ext.c
+++ b/test/v3ext.c
@@ -37,6 +37,7 @@ static int test_pathlen(void)
     return ret;
 }
 
+#ifndef OPENSSL_NO_RFC3779
 static int test_asid(void)
 {
     ASN1_INTEGER *val1 = NULL, *val2 = NULL;
@@ -113,6 +114,7 @@ static int test_asid(void)
     ASIdentifiers_free(asid4);
     return testresult;
 }
+#endif /* OPENSSL_NO_RFC3779 */
 
 OPT_TEST_DECLARE_USAGE("cert.pem\n")
 
@@ -127,6 +129,8 @@ int setup_tests(void)
         return 0;
 
     ADD_TEST(test_pathlen);
+#ifndef OPENSSL_NO_RFC3779
     ADD_TEST(test_asid);
+#endif /* OPENSSL_NO_RFC3779 */
     return 1;
 }
